### ISSUE TITLE
Critical: somehow editing and saving a layout that contains a "LW Subsystem" (Live Window Susbsystem) section may result in corrupted layout

### ISSUE THREAD

**Issue by gartaud** [wpilibsuite/SmartDashboard#101]
Symptoms:
Open a perfectly good layout. Accept to save layout upon exiting Smart Dash. Try opening layout again. Smart Dash crashes.
Reason for crash: the layout is incomplete (XML file is not properly terminated).
Note: It would be nice to get a dialog explaining that the XML file is corrupted...

Reason for layout to be corrupted after saving:
DashBoardFrame.saveElements() expect all elements to have a non-null type, and queries the name of the type without making sure that the type is not null when calling:
```
writer.beginWidget(((Widget) element).getFieldName(), ((Widget) element).getType()
            .getName(), element.getClass().getName());
```
Somehow that is not always true (at least for LW Subsystem)

Adding a few println prior to that line prove it:
```
System.out.println("field name: " + ((Widget) element).getFieldName());
System.out.println("class name: " + element.getClass().getName());
if (element instanceof LWSubsystem) {
	System.out.println("LWSubsystem!");
}
System.out.println("type: " + ((Widget) element).getType());
System.out.println("type name: " + ((Widget) element).getType().getName());
```
OUTPUT:
```
field name: Ungrouped
class name: edu.wpi.first.smartdashboard.livewindow.elements.LWSubsystem
LWSubsystem!
type: null

Mar 01, 2018 8:13:33 PM edu.wpi.first.smartdashboard.gui.DashboardFrame save
SEVERE: null
java.lang.NullPointerException
	at edu.wpi.first.smartdashboard.gui.DashboardFrame.saveElements(DashboardFrame.java:289)
	at edu.wpi.first.smartdashboard.gui.DashboardFrame.save(DashboardFrame.java:263)
	at edu.wpi.first.smartdashboard.gui.DashboardFrame.exit(DashboardFrame.java:425)
	at edu.wpi.first.smartdashboard.gui.DashboardFrame$2.windowClosing(DashboardFrame.java:180)
```
Please handle properly scenario where type is somehow null. Thanks!

**Comment by gartaud** [comment]
Hi @JLLeitschuh 
Can we expect this to be fixed soon?
Thanks!

**Comment by gartaud** [comment]
Thanks @PeterJohnson !

**Comment by nathan815** [comment]
Whew, we ran across this at our first competition right before a match which caused a bit of stress (without SmartDashboard we can't choose our autonomous mode). Had to temporarily switch to ShuffleBoard which has been very buggy in our experience (and for some reason required restarting on the field before many matches in our case).

Glad to see this was fixed.

I may submit a PR in the future to try to handle uncaught exceptions like this, instead of just crashing.
